pkg/store/tsdb.go

```diff
@@ -237,6 +245,24 @@ func (s *TSDBStore) SeriesLocal(ctx context.Context, r *storepb.SeriesRequest) (
 	return rs.series, nil
 }
 
+// MatchersToPromMatchers converts storepb label matchers to prometheus label matchers. It goes to cache if matcherConverter is set.
+func (s *TSDBStore) MatchersToPromMatchers(ms ...storepb.LabelMatcher) ([]*labels.Matcher, error) {
```

Comment:
this logic is redunant with ones in `ProxyStore`, consider to just pass matcherConverter as a func arg to `matchesExternalLabels`?

```
func (s *ProxyStore) MatchersToPromMatchers
```

Reply:
+1